Make the filter name configurable independently of the attributes

cmlenz opened the pull request:
We had a case where we needed to use a custom filter, but continue to use the regular `angular-gettext` directive. So we need to configure the filter name independently of the attributes. This PR does that.

Projects using a custom attribute and filter will need to set the `filterName` option accordingly, so this is a breaking changes.

rubenv commented:
> Projects using a custom attribute and filter will need to set the `filterName` option accordingly, so this is a breaking changes.

That makes it pretty much a no-no. No breaking changes allowed anymore.

Is there any way to make this non-breaking?

cmlenz commented:
Oh, I didn't expect you were still reviewing the PRs here. I created it more for the record. Sorry.

If there's a chance that changes still get integrated, I would put more work into the PR to make it a non-breaking change.

rubenv commented:
If it's purely additive and non-breaking, sure, why not.

cmlenz commented:
I have updated the PR to make this a non-breaking change as requested.

If you don't specify the new `filterName` option, the extractor will continue to use `options.attribute` just like before. It's only when `options.filterName` is set that it will be used instead.

Also, I have added new tests instead of changing the existing ones.

rubenv commented:
Released to NPM as `v2.5.2`

cmlenz commented:
Thank you for the very quick release!
